packages/@expo/prebuild-config/src/plugins/unversioned/expo-splash-screen/withAndroidSplashImages.ts

```diff
@@ -149,28 +149,33 @@ export async function setSplashImageDrawablesForThemeAsync(
   if (!config) return;
   const androidMainPath = path.join(projectRoot, 'android/app/src/main');
 
+  const sizes: DRAWABLE_SIZE[] = ['mdpi', 'hdpi', 'xhdpi', 'xxhdpi', 'xxxhdpi'];
+
   await Promise.all(
-    ['mdpi', 'hdpi', 'xhdpi', 'xxhdpi', 'xxxhdpi'].map(async (imageKey) => {
+    sizes.map(async (imageKey) => {
       // @ts-ignore
-      const image = config[imageKey];
+      const image = await Jimp.read(config[imageKey]);
+
       if (image) {
-        // Using this method will cache the images in `.expo` based on the properties used to generate them.
-        // this method also supports remote URLs and using the global sharp instance.
-        const { source } = await generateImageAsync({ projectRoot, cacheType: IMAGE_CACHE_NAME }, {
-          src: image,
-        } as any);
+        const multiplier = DRAWABLES_CONFIGS[imageKey].dimensionsMultiplier;
+
+        // https://developer.android.com/develop/ui/views/launch/splash-screen#dimensions
+        const canvasSize = 288 * multiplier;
+        const canvas = await Jimp.create(canvasSize, Jimp.AUTO, '#ffffff');
```

Comment:
Set `canvasSize` as second argument too (to avoid out of range crashes).
For the moment, the image is not correctly generated. It's not transparent + not centered:

![splashscreen_logo](https://github.com/user-attachments/assets/dbf79765-dff0-41a5-9649-838a3a083c6c)

Try this:

```ts
const multiplier = DRAWABLES_CONFIGS[imageKey].dimensionsMultiplier;

const width = 100 * multiplier; // "100" must be replaced by the logo width chosen by the user in its config file
const height = Math.ceil(width * (image.bitmap.height / image.bitmap.width));

// https://developer.android.com/develop/ui/views/launch/splash-screen#dimensions
const canvasSize = 288 * multiplier;
const canvas = await Jimp.create(canvasSize, canvasSize, 0xffffff00);
const input = image.clone().resize(width, height);

const x = (canvasSize - width) / 2;
const y = (canvasSize - height) / 2;

const output = canvas.blit(input, x, y).quality(100);

// …

Reply:
This didn't work. The images are stretched. 

This works, just need to accept the image size
```typescript
    const canvasSize = 288 * multiplier;
    const canvas = await Jimp.create(canvasSize, canvasSize, 0xffffff00);
    const input = image.clone().resize(100 * multiplier, 100 * multiplier);
    const output = canvas.composite(input, 100 * multiplier, 100 * multiplier);
```

Reply:
Could this be linked to the fact that `jimp-compact` is several versions behind `jimp`? I didn't tried it, but:

```ts
const input = image.clone().resize(100 * multiplier, 100 * multiplier);
```

If you set the height to be identical as the width (`100 * multiplier`), I'm afraid this will only work for square images (like the current one). You can try with this instead: https://github.com/zoontek/react-native-bootsplash/blob/4.4.2/example/assets/bootsplash_logo_original.png (100x89)